cmake/Modules/FindHWLOC.cmake

```diff
@@ -0,0 +1,147 @@
+#
```

Comment:
Is there any particular reason you prefer this over the current one ?

Reply:
Licensing. The license in this one is close to our own and more permissive than the LGPL. Since this is a core aspect of our library, we would pretty much always have it as part of our package, which could create future trouble.

I also know personally where this is from and used it myself so I can attest that it is solid. But it of course depends if it does all we need.

Reply:
I use this now but modify it for our purposes.